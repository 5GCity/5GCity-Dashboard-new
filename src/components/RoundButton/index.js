/**
 * RoundButton Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import React from 'react'
import styled from 'styled-components'
import { darken } from 'polished';

export default ({ children, icon, ...props }) => (
  <RoundButton>
    <IconContent>{icon}</IconContent>
  </RoundButton>
)

const RoundButton = styled.div`

  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  ${({size}) => size ? `
  height: ${size}px;
  width: ${size}px;
  `:`
  height: 32px;
  width: 32px;
`}
  border-radius: 24px;
  background-color: #89979F;

  &:hover{
    background-color: ${darken(0.2,'#89979F')};
  }

`
const IconContent = styled.i`
`

