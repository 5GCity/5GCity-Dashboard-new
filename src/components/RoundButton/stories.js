/**
 * RoundButton Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'
import { Theme } from 'globalStyles'

import RoundButton from './index'

const bStyle = {
  padding: 30,
  backgroundColor: Theme.bodyBackground,
  height: '100%'
}

const exampleOne = () => (
  <div style={bStyle}>
    <RoundButton color={'#fff'} />
  </div>
)

storiesOf('RoundButton', module)
  .add('primary', exampleOne)
