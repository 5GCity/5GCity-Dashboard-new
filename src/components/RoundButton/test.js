/**
 * RoundButton Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import RoundButton from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<RoundButton />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
