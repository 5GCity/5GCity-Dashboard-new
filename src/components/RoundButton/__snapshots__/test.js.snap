// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  height: 32px;
  width: 32px;
  border-radius: 24px;
  background-color: #89979F;
}

.c0:hover {
  background-color: #57646b;
}

<div
  className="c0"
  size={undefined}
>
  <i
    className=""
  />
</div>
`;
